### PR TITLE
Listing page - basic Listing page

### PR DESCRIPTION
**Pull request by elinaKM**
Functionality missing: animal type should be passed by props. Need to be done.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,13 +3,14 @@ import { Router } from "@reach/router";
...
 import PetPage from "./pages/PetPage";
+import SearchPage from "./pages/SearchPage";
 import NotFound from "./pages/NotFound";
 import Header from "./components/Header";
 import Footer from "./components/Footer";
 import SideDrawerMenu from "./components/sideMenu/SideDrawerMenu";
 import Backdrop from "./components/common/Backdrop";
 import "./index.css";
-import { ROOT, PET } from "./configuration/pathes";
+import { ROOT, PETID, PETS } from "./configuration/pathes";
```
Comment (okvv) [inline src/App.js]: PET_ID

File: src/components/PetCard.js [hunk 2]
```diff
@@ -3,14 +3,17 @@ import styled from 'styled-components'
...
-const PetCard = ({ name, img, id }) => (
-    <StyledLink to={id ? PETS + id : PETS} >
-        <Image img={img} />
-        <Name>
-            {name}
-        </Name>
-    </StyledLink>
-)
+const PetCard = ({name, imgs, id, fromPreview}) => {
+    let path = id ? (fromPreview ? PETS + id :id) : PETS;
```
Comment (okvv) [inline src/components/PetCard.js]: why let?

File: src/components/PetListing.js [hunk 3]
```diff
@@ -2,18 +2,19 @@ import React from 'react'
 import styled from 'styled-components'
 import PetCard from './PetCard'
 
-const PetListing = ({pets}) => (
+const PetListing = ({pets, fromPreview}) => (
     <Wrapper>
         {
             pets.map((pet) =>
-            <PetCard name={pet.name} key={pet.id} img={pet.img} id={pet.id}/>
+            <PetCard name={pet.name} key={pet.id} imgs={pet.imgs} id={pet.id} fromPreview={fromPreview}/>
```
Comment (okvv) [inline src/components/PetListing.js]: fromPreview? what's that for?

File: src/components/PetListingPreview.js [hunk 4]
```diff
@@ -4,17 +4,20 @@ import catSteps from '../images/paw-print.jpg'
...
-            Pets Available for Adoption
-        </Title>
-        <ListingWrapper>
-            <PetListing pets={pets} />
-            <MeetMoreButton name={"Meet more friends"} img={catSteps}/>
-        </ListingWrapper>
-    </Wrapper>
-)
+const PetListingPreview = ({pets}) => {
+    let imagesArr = [catSteps];
```
Comment (okvv) [inline src/components/PetListingPreview.js]: why let?
what's catSteps?

File: src/configuration/pathes.js [hunk 5]
```diff
@@ -1,3 +1,3 @@
 export const ROOT="petfinder/"
 export const PETS="pets/"
-export const PET="pets/:petId"
\ No newline at end of file
+export const PETID=":petId"
```
Comment (okvv) [inline src/configuration/pathes.js]: PET_ID

File: src/pages/PetPage.js [hunk 6]
```diff
@@ -1,102 +1,17 @@
...
-        org: "CARMA, Fredericton Chapter"
-    }
-}
+import { petsDB } from './../petsDB'
+import find from "lodash/find"
 
 const PetPage = (props) => {
 
-    const propObject = petsDB.hasOwnProperty(props.petId) ? petsDB[props.petId] : petsDB[1];
+    let propObject = find(petsDB, ['id', props.petId]);
```
Comment (okvv) [inline src/pages/PetPage.js:9]: why let?

File: src/components/PetListingPreview.js [hunk 7]
```diff
@@ -4,17 +4,20 @@ import catSteps from '../images/paw-print.jpg'
...
+    return(
+        <Wrapper>
+            <Title>
+                Pets Available for Adoption
+            </Title>
+            <ListingWrapper>
+                <PetListing pets={pets} fromPreview/>
+                <MeetMoreButton name={"Meet more friends"} imgs={imagesArr}/>
```
Comment (okvv) [inline src/components/PetListingPreview.js]: name={"Meet more friends"}
you're passing what JS type exactly?

File: package.json [hunk 8]
```diff
@@ -7,9 +7,11 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "lodash.find": "^4.6.0",
```
Comment (okvv) [inline package.json]: please use lodash and import needed functions

**Review by okvv — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (elinaKM) [reply src/App.js]: done

File: src/components/PetCard.js (hunk 2 above)
Comment (elinaKM) [reply src/components/PetCard.js]: path and link was removed

File: src/components/PetListing.js (hunk 3 above)
Comment (elinaKM) [reply src/components/PetListing.js]: fromPreview removed

File: src/components/PetListingPreview.js (hunk 4 above)
Comment (elinaKM) [reply src/components/PetListingPreview.js]: removed

File: src/configuration/pathes.js (hunk 5 above)
Comment (elinaKM) [reply src/configuration/pathes.js]: done

File: src/pages/PetPage.js (hunk 6 above)
Comment (elinaKM) [reply src/pages/PetPage.js:9]: stay let, because it's being updated

File: src/components/PetListingPreview.js (hunk 7 above)
Comment (elinaKM) [reply src/components/PetListingPreview.js]: fixed

File: package.json (hunk 8 above)
Comment (elinaKM) [reply package.json]: done

File: src/pages/SearchPage.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+const SearchPage = (props) => {
+    const [type, setType] = useState("Animal");
+    const [breeds, setBreeds] = useState([]);
+    const [age, setAge] = useState([]);
+    const [size, setSize] = useState([]);
+    const [gender, setGender] = useState();
+
+    const getStringValues = (value) => {
+        return value == null ? [] : value.map((item) => item.value);
+    }
```
Comment (okvv) [inline src/pages/SearchPage.js]: tight coupling with LabledSelect
are you sure that's the way to pass values around?
you've got nested calls in your onChange

**Review by okvv — CHANGES_REQUESTED** [review]

File: src/pages/SearchPage.js (hunk 9 above)
Comment (elinaKM) [reply src/pages/SearchPage.js]: Fixed

File: src/components/common/LabeledSelect.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import styled from 'styled-components'
+import Select from 'react-select'
+import find from "lodash/find"
+
+const getStringValues = (value) => {
+    return value == null ? [] : value.map((item) => item.value);
```
Comment (okvv) [inline src/components/common/LabeledSelect.js]: === null

Comment (okvv) [inline src/components/common/LabeledSelect.js]: use lodash map
map(values, 'value')

**Review by okvv — CHANGES_REQUESTED** [review]

File: src/components/common/LabeledSelect.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+const LabeledSelect = ({label, value, options, onChange, ...rest}) => (
```
Comment (okvv) [inline src/components/common/LabeledSelect.js]: add isMulti if you're using it and add a default value

File: src/components/common/LabeledSelect.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    <Wrapper>
+        <label>{label}</label>
+        <StyledSelect
+            placeholder="Any"
+            value={value ? find(options, { value }) : null}
+            onChange={(value) => onChange( rest.isMulti ? getStringValues(value) : value)}
```
Comment (okvv) [inline src/components/common/LabeledSelect.js]: isMulti

**Review by okvv — CHANGES_REQUESTED** [review]

File: src/components/common/LabeledSelect.js (hunk 12 above)
Comment (okvv) [reply src/components/common/LabeledSelect.js]: onChange should call a function that handles the isMulti logic and has a reference to a parent function to update state (if needed)

**Review by okvv — CHANGES_REQUESTED** [review]

File: src/components/common/LabeledSelect.js (hunk 10 above)
Comment (elinaKM) [reply src/components/common/LabeledSelect.js]: fixed

Comment (elinaKM) [reply src/components/common/LabeledSelect.js]: done

File: src/components/common/LabeledSelect.js (hunk 11 above)
Comment (elinaKM) [reply src/components/common/LabeledSelect.js]: done, no default required, if not set - not Multi

File: src/components/common/LabeledSelect.js (hunk 12 above)
Comment (elinaKM) [reply src/components/common/LabeledSelect.js]: done

File: src/components/common/LabeledSelect.js (hunk 11 above)
Comment (okvv) [reply src/components/common/LabeledSelect.js]: well it's a boolean, better to default to false

File: src/components/common/LabeledSelect.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    }
+}
+
+const LabeledSelect = ({label, value, options, onChangeHandler, isMulti, ...rest}) => (
+    <Wrapper>
+        <label>{label}</label>
+        <StyledSelect
+            placeholder="Any"
+            value={value ? find(options, { value }) : null}
+            onChange={(values) => onChangeHandler(getStringValues(values, isMulti))}
```
Comment (okvv) [inline src/components/common/LabeledSelect.js]: discussed

File: src/pages/SearchPage.js [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+                        onChangeHandler={setGender}
+                    />
+                </Selectors>
+            </SidePanel>
+            <ListingContainer>
+                <Title>
+                    Available {type}s for Adoption:
+                </Title>
+                <ListingWrapper>
+                    {petsDB.map((pet) =>
```
Comment (okvv) [inline src/pages/SearchPage.js]:
```suggestion
                    {petsDB.map(({name, id, imgs}) =>
```

File: src/pages/SearchPage.js [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+                        <StyledLink to={pet.id}>
+                            <PetCard name={pet.name} key={pet.id} imgs={pet.imgs} id={pet.id} />
```
Comment (okvv) [inline src/pages/SearchPage.js]:
```suggestion
                        <StyledLink to={id}>
                            <PetCard name={name} key={id} imgs={imgs} id={id} />
```

File: src/components/common/LabeledSelect.js [hunk 16]
```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import styled from 'styled-components'
+import Select from 'react-select'
+import find from "lodash/find"
+import map from "lodash/map"
+
+const getStringValues = (values=[], isMulti="false") => (
```
Comment (okvv) [inline src/components/common/LabeledSelect.js]: isMulti="false"? we're passing strings for booleans now?

File: src/components/common/LabeledSelect.js [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+    const onSelectValueChange = (value, isMulti) => onChangeHandler(getStringValues(value, isMulti));
+
+    return (
+        <Wrapper>
+            <label>{label}</label>
+            <StyledSelect
+                placeholder="Any"
+                value={value ? find(options, { value }) : null}
+                onChange={onSelectValueChange}
```
Comment (okvv) [inline src/components/common/LabeledSelect.js]: discussed

**Review by okvv — CHANGES_REQUESTED** [review]

File: src/components/common/LabeledSelect.js (hunk 16 above)
Comment (elinaKM) [reply src/components/common/LabeledSelect.js]: :)))) nope...

File: src/pages/SearchPage.js (hunk 14 above)
Comment (elinaKM) [reply src/pages/SearchPage.js]: fixed, also in PetListingPreview Component

File: src/pages/SearchPage.js (hunk 15 above)
Comment (elinaKM) [reply src/pages/SearchPage.js]: fixed

File: src/components/common/LabeledSelect.js [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+import map from "lodash/map"
+
+const LabeledSelect = ({ label, value, options, onChangeHandler, isMulti, ...rest }) => {
+
+    const onSelectHandler = (value=[]) => {
+        if (Array.isArray(value)) {
+            onChangeHandler(map(value, (item) => item.value))
+        } else {
+            onChangeHandler(value.value)
+        }
```
Comment (okvv) [inline src/components/common/LabeledSelect.js]: discussed

**Review by okvv — CHANGES_REQUESTED** [review]

**Review by okvv — APPROVED** [review]

**Pull request merged**